emukit/core/constraints.py

```diff
@@ -0,0 +1,97 @@
+import logging
+from typing import Callable, Optional
+
+import numpy as np
+
+_log = logging.getLogger(__name__)
+
+
+class IConstraint:
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        :param x: Array of shape (n_points x n_dims) containing input locations to evaluate constraint at
+        :return: Numpy array of shape (n_input,) where an element will be 1 if the corresponding input satisfies the
+                 constraint and zero if the constraint is violated
+        """
+        raise NotImplementedError
+
+
+class LinearInequalityConstraint(IConstraint):
+    """
+    Constraint of the form lower_bound < Ax < upper_bound where the matrix A is called "constraint_matrix"
+    """
+    def __init__(self, constraint_matrix: np.ndarray, lower_bound: np.ndarray=None, upper_bound: np.ndarray=None):
+        """
+
+        :param constraint_matrix: (n_constraint, n_x_dims) matrix in b_lower < Ax < b_upper
+        :param lower_bound: Lower bound vector of size (n_constraint,). Can be -np.inf for one sided constraint
+        :param upper_bound: Upper bound vector of size (n_constraint,). Can be np.inf for one sided constraint
+        """
+        if (lower_bound is None) and (upper_bound is None):
```

Comment:
You may also want to make sure they are not +-inf at the same time, maybe after defaults are handled

Reply:
Added that validation below